### PR TITLE
 RDM-3079: Search by case id not working with hyphens

### PR DESCRIPTION
**Pull request by Thor-tech-of-metal**
RDM-3079: Search by case id not working with hyphens
      [RDM-3079] (https://tools.hmcts.net/jira/browse/RDM-3079).

### CODE REVIEW

File: src/aat/java/uk/gov/hmcts/ccd/datastore/util/CaseIdHelper.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    private CaseIdHelper() {
+    }
+
+    public static String hypheniseACaseId(final String caseId) {
+
+        if (caseId.length() == 16) {
+            final String[] strSubstrings = caseId.split("(?<=\\G.{4})");
+            return strSubstrings[0] + "-" + strSubstrings[1] + "-" + strSubstrings[2] + "-" + strSubstrings[3];
+        } else {
+            throw new FunctionalTestException("Problem getting case id as hypensedFormat");
```
Comment (kiran-yenigala-hmcts) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/util/CaseIdHelper.java:16]: this error seems to be incorrect, there is no problem in getting hypensed format, but we have an invalid case id

File: src/aat/java/uk/gov/hmcts/ccd/datastore/util/CaseIdHelper.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+public class CaseIdHelper {
+
+    private CaseIdHelper() {
+    }
+
+    public static String hypheniseACaseId(final String caseId) {
+
+        if (caseId.length() == 16) {
+            final String[] strSubstrings = caseId.split("(?<=\\G.{4})");
+            return strSubstrings[0] + "-" + strSubstrings[1] + "-" + strSubstrings[2] + "-" + strSubstrings[3];
```
Comment (kiran-yenigala-hmcts) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/util/CaseIdHelper.java:14]: String.join("-", strSubstrings); --> better to use this

**Review by kiran-yenigala-hmcts — COMMENTED** [review]
minor comments

**Review by kiran-yenigala-hmcts — APPROVED** [review]

**Review by mario-paniccia — APPROVED** [review]
👍

File: Jenkinsfile_CNP [hunk 3]
```diff
@@ -115,7 +115,7 @@ withPipeline(type, product, component) {
             environmentOfDependencies = "aat"
         }
         env.DEFINITION_STORE_HOST      = "http://ccd-definition-store-api-${environmentOfDependencies}.service.core-compute-${environmentOfDependencies}.internal"
-        env.ELASTIC_SEARCH_FTA_ENABLED = "true"
+        env.ELASTIC_SEARCH_FTA_ENABLED = "false"
     }
```
Comment (Thor-tech-of-metal) [inline Jenkinsfile_CNP:119]: By default ES tests are ignored on Preview envs but should run on aat/master.

File: Jenkinsfile_CNP [hunk 4]
```diff
@@ -98,7 +98,7 @@ withPipeline(type, product, component) {
...
     if (env.BRANCH_NAME.equalsIgnoreCase(dataStoreApiDevelopPr)) {
         enableSlackNotifications('#ccd-develop-builds')
-        env.ELASTIC_SEARCH_FTA_ENABLED = "true"
+        env.ELASTIC_SEARCH_FTA_ENABLED = "false"
     }
```
Comment (Thor-tech-of-metal) [inline Jenkinsfile_CNP:102]: That value has fixed build issue.

**Pull request merged**